azurerm/resource_arm_automation_runbook.go

```diff
@@ -0,0 +1,260 @@
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"algorithm": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"value": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmAutomationRunbookCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).automationRunbookClient
+	log.Printf("[INFO] preparing arguments for AzureRM Automation Runbook creation.")
+
+	name := d.Get("name").(string)
+	location := d.Get("location").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	tags := d.Get("tags").(map[string]interface{})
+
+	accName := d.Get("account_name").(string)
+	runbookType := automation.RunbookTypeEnum(d.Get("runbook_type").(string))
+	logProgress := d.Get("log_progress").(bool)
+	logVerbose := d.Get("log_verbose").(bool)
+	description := d.Get("description").(string)
+
+	contentLink := expandContentLink(d)
+
+	parameters := automation.RunbookCreateOrUpdateParameters{
+		RunbookCreateOrUpdateProperties: &automation.RunbookCreateOrUpdateProperties{
+			LogVerbose:         &logVerbose,
+			LogProgress:        &logProgress,
+			RunbookType:        runbookType,
+			Description:        &description,
+			PublishContentLink: &contentLink,
+		},
+
+		Location: &location,
+		Tags:     expandTags(tags),
+	}
+
+	_, err := client.CreateOrUpdate(resGroup, accName, name, parameters)
+	if err != nil {
+		return err
+	}
+
+	read, err := client.Get(resGroup, accName, name)
+	if err != nil {
+		return err
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read Automation Runbook '%s' (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmAutomationRunbookRead(d, meta)
+}
+
+func resourceArmAutomationRunbookRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).automationRunbookClient
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	accName := id.Path["automationAccounts"]
+	name := id.Path["runbooks"]
+
+	resp, err := client.Get(resGroup, accName, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error making Read request on AzureRM Automation Runbook '%s': %+v", name, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("location", azureRMNormalizeLocation(*resp.Location))
+	d.Set("resource_group_name", resGroup)
+
+	d.Set("account_name", accName)
+	if props := resp.RunbookProperties; props != nil {
+		d.Set("log_verbose", props.LogVerbose)
+		d.Set("log_progress", props.LogProgress)
+		d.Set("runbook_type", props.RunbookType)
+		d.Set("description", props.Description)
+	}
+
+	d.Set("publish_content_link", nil) //publish content link is not set during Get()
```

Comment:
from what I can see, [this should be available in the response](https://docs.microsoft.com/en-us/rest/api/automation/runbook/get#definitions_runbook_properties_publishcontentlink) - is there a reason we're not setting this?

Reply:
That was my first tought too but as you can see on the site you linked the publishContentLink is not listed in the example JSON. If i dump the result object of the Get it is also nil:
```
&automation.RunbookProperties{RunbookType:"PowerShellWorkflow", PublishContentLink:(*automation.ContentLink)(nil), State:"Published", LogVerbose:(*bool)(0xc420389168), LogProgress:(*
bool)(0xc420389169), LogActivityTrace:(*int32)(0xc42038916c), JobCount:(*int32)(0xc4203891fc), Parameters:(*map[string]*automation.RunbookParameter)(0xc4202b83c0), OutputTypes:(*[]string)(0xc4204c41e0), Draft:(*automation.RunbookDraft)(nil), ProvisioningState:"Succeeded"
, LastModifiedBy:(*string)(0xc420389260), CreationTime:(*date.Time)(0xc4204c4220), LastModifiedTime:(*date.Time)(0xc4204c4320), Description:(*string)(0xc4203891e0)}
```

In this case should i remove the nil association?

Reply:
I removed, now its cleaner because it doesn't cause perpetual diff. As soon as it will be available in the response we can put it back.